storage/remote/queue_manager.go

```diff
@@ -1026,6 +1031,9 @@ func (s *shards) enqueue(ref chunks.HeadSeriesRef, data sampleOrExemplar) bool {
 type queue struct {
 	batch      []sampleOrExemplar
 	batchQueue chan []sampleOrExemplar
+	// batchMtx covers sending to the batchQueue and batch operations other
+	// than appending a sample.
+	batchMtx sync.Mutex
```

Comment:
nit, can we keep mtx fields above the things they lock in the struct field listings

Reply:
:+1: I also change the other mutex for queue to be consistent.